Components/FlightControl/WatchdogTask.cpp

```diff
@@ -0,0 +1,123 @@
+/**
+ ******************************************************************************
+ * File Name          : WatchdogTask.cpp
+ * Description        : Primary flight task, default task for the system.
+ ******************************************************************************
+*/
+#include "GPIO.hpp"
+#include "SystemDefines.hpp"
+#include "Timer.hpp"
+#include "WatchdogTask.hpp"
+#include "FlightTask.hpp"
+
+
+/**
+ * @brief Initialize the WatchdogTask
+ * @params Must pass in the timer period, If no heartbeat in received within this period then the timerr will be reset
+ */
+WatchdogTask::WatchdogTask(){}
+
+/**
+ * @brief Initialize the FlightTask
+ */
+void WatchdogTask::InitTask()
+{
+    // Make sure the task is not already initialized
+    SOAR_ASSERT(rtTaskHandle == nullptr, "Cannot initialize flight task twice");
```

Comment:
Cannot initialize **watchdog** task twice

(I know there's a few renames I missed in some other tasks too though 😓)